### PR TITLE
Yarik sagotly fab

### PR DESCRIPTION
**Pull request by YaroslavNevsky**
dont hate me if i make a shit please............

### CODE REVIEW

**Comment by sagotly** [comment]
mostly sure have some bugs in it. Proper bug test needed

File: handlers/directory_handler.go [hunk 1]
```diff
@@ -55,34 +60,34 @@ func DirectoryHandler(c *gin.Context) {
...
 	}
-	fmt.Println("current path" + requestPath)
-	fmt.Println("root path" + RootDirectoryPath)
 
 	fileInfos, err := directory.Readdir(-1)
 	if err != nil {
 		c.JSON(500, gin.H{"error": err.Error()})
 		return
 	}
+	//get uploaded files
```
Comment (fenix1851) [inline handlers/directory_handler.go]: what is this? @sagotly

File: handlers/directory_handler.go [hunk 2]
```diff
@@ -122,9 +127,21 @@ func DirectoryHandler(c *gin.Context) {
...
+		return files[i].FileName < files[j].FileName
+	})
+	//sort directories
+	sort.Slice(directories, func(i, j int) bool {
+		return strings.ToLower(directories[i].DirectoryName) < strings.ToLower(directories[j].DirectoryName)
+	})
 
 	// Render the template or return JSON data
 	if len(directories) > 0 || len(files) > 0 {
+		//clear cache
```
Comment (fenix1851) [inline handlers/directory_handler.go:132]: nice, this is actually good)

File: handlers/directory_handler.go [hunk 3]
```diff
@@ -141,3 +158,39 @@ func FileHandler(c *gin.Context) {
...
+	file, err := c.FormFile("file")
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	absUrl := c.PostForm("absUrl")
+	fmt.Print("\nABSOLUTE URL:", absUrl)
+	parsedURL, err := url.Parse(absUrl)
+	if err != nil {
+		fmt.Println("Ошибка разбора URL:", err)
```
Comment (fenix1851) [inline handlers/directory_handler.go]: return some responce, maybe 400

File: handlers/directory_handler.go [hunk 4]
```diff
@@ -141,3 +158,39 @@ func FileHandler(c *gin.Context) {
...
+
+	// getting
+	fileName := fmt.Sprintf("id_%s_%s", uuid.New().String()[:8], file.Filename)
+	fullPath := filepath.Join(path, fileName)
+
+	fmt.Println("\nfull path:", fullPath)
+	fmt.Println("current Dir Path: ", path)
+
+	err = c.SaveUploadedFile(file, fullPath)
+	fmt.Println("\nkonec________________________________")
```
Comment (fenix1851) [inline handlers/directory_handler.go]: what is this language?)

File: handlers/directory_handler.go [hunk 5]
```diff
@@ -141,3 +158,39 @@ func FileHandler(c *gin.Context) {
...
+	if err != nil {
+		fmt.Printf("error has accured:%s", err)
```
Comment (fenix1851) [inline handlers/directory_handler.go]: occurred

File: templates/directory.html [hunk 6]
```diff
@@ -31,19 +32,19 @@
...
         align-items: center;
-        margin-bottom: 10px;
+        margin-bottom: 0.625rem; 
         cursor: pointer;
         transition: background-color 0.2s ease-in-out;
     }
 
     .item:hover {
         background-color: #292757;
-        /* Голубовато-синеватое свечение */
```
Comment (fenix1851) [inline templates/directory.html:41]: thx

File: templates/directory.html [hunk 7]
```diff
@@ -95,20 +220,68 @@ <h1>Directory Listing: {{.FolderName}}</h1>
     <ul>
         {{range .Directories}}
         <li class="item">
-            <span class="icon">📁</span> <!-- Значок папки -->
```
Comment (fenix1851) [inline templates/directory.html:98]: thx

File: templates/directory.html [hunk 8]
```diff
@@ -95,20 +220,68 @@ <h1>Directory Listing: {{.FolderName}}</h1>
...
+            <span class="icon">📁</span>
             <a class="directory" href="{{.DirectoryPath}}">{{.DirectoryName}}</a>
         </li>
         {{end}}
         {{range .Files}}
         <li class="item">
-            <span class="icon">📄</span> <!-- Значок файла -->
```
Comment (fenix1851) [inline templates/directory.html:104]: thx

File: templates/directory.html [hunk 9]
```diff
@@ -95,20 +220,68 @@ <h1>Directory Listing: {{.FolderName}}</h1>
...
+            <span class="icon">📄</span>
             <a class="file" href="{{.FilePath}}">{{.FileName}}</a>
         </li>
         {{end}}
     </ul>
-    <footer>
-        <a class="github-link" href="https://github.com/fenix1851/go-file-server">GitHub Repository</a>
+    <footer class="github-link">
+        <a style="color: #d6e2e3; text-decoration: none; font-weight: bold;  font-family: 'Ubuntu', sans-serif;" href="https://github.com/fenix1851/go-file-server">GitHub Repository</a>
```
Comment (fenix1851) [inline templates/directory.html]: ??? why all this styles are inside attribute?

File: templates/directory.html [hunk 10]
```diff
@@ -95,20 +220,68 @@ <h1>Directory Listing: {{.FolderName}}</h1>
...
+                    .catch(error => {
+                       
+                        console.error('Error:', error);
+                    });
+                }
+                location.reload(); // Перезагрузка страницы
+            });
+        });
+        
+        const mobileMediaQuery = window.matchMedia("(max-width: 768px)");
```
Comment (fenix1851) [inline templates/directory.html:283]: 🔥🔥🔥🔥🔥

**Review by fenix1851 — CHANGES_REQUESTED** [review]

File: handlers/directory_handler.go (hunk 1 above)
Comment (fenix1851) [reply handlers/directory_handler.go]: check code below 76 line

File: handlers/directory_handler.go (hunk 3 above)
Comment (fenix1851) [reply handlers/directory_handler.go]: check code below 173 line

File: handlers/directory_handler.go (hunk 4 above)
Comment (sagotly) [reply handlers/directory_handler.go]: xD

File: handlers/directory_handler.go [hunk 11]
```diff
@@ -18,7 +22,7 @@ func DirectoryHandler(c *gin.Context) {
 	var CurrentDirectoryPath string = startup.RootPath
 	fmt.Println(requestPath, "requestPath")
 	if requestPath != "/" {
-		// Преобразуем путь в стиле Windows в путь в стиле Unix
+		// Convert a Windows-style path to a Unix-style path
```
Comment (fenix1851) [inline handlers/directory_handler.go:25]: good)

**Review by fenix1851 — APPROVED** [review]
it's ok)

**Pull request merged**